node-graph/gstd/src/gpu_nodes.rs

```diff
@@ -392,7 +392,7 @@ pub struct BlendGpuImageNode<Background, B, O> {
 }
 
 #[node_macro::node_fn(BlendGpuImageNode)]
-async fn blend_gpu_image(foreground: ImageFrame<Color>, background: ImageFrame<Color>, blend_mode: BlendMode, opacity: f32) -> ImageFrame<Color> {
+async fn blend_gpu_image(foreground: ImageFrame<Color>, background: ImageFrame<Color>, blend_mode: BlendMode, opacity: f32) -> Result<ImageFrame<Color>, String> {
 	let foreground_size = DVec2::new(foreground.image.width as f64, foreground.image.height as f64);
```

Comment:
At this point we probably don't want to return a result because this is part of the node system which would force the user to deal with all of the error management themselves. In this case I think it would make most sense to throw a console warning and then return an empty image frame

Reply:
Thanks very much for your comments, I will incorporate these! Quick question though on this one - when you say console message do you mean a `log::debug!` macro call, or something more specific where we send an explicit `ErrorDialog` to the frontend?

Reply:
sending an explict dialouge to the frontend would of course be ideal but that does require extra work. For now you can use a `log::error!`

Reply:
I'd rather we don't annoy the user with a dialog anytime they create a cycle, that connection should just be silently rejected.

Reply:
Yes in that case the dialogue won't be displayed because the cycle never exists. This is intended to cover cases where the user loads an invalid document for example

Reply:
Makes sense - thanks both!

Reply:
I've updated this one - it works but I'm not sure if it's not 100% idiomatic Rust, so let me know if you would like me to change it. I tried using a match statement or chaining functions but ultimately we have an expression there that has to return a different type if it's an error, so I just used a simple if statement.